S2024 dynamics add acceleration calibration utility to utils

Andy-Chen-design opened the pull request:
- Implemented functions to extract measured and output acceleration, as well as speed data from txt files.
- Added friction modeling and curve fitting to calculate and plot friction forces based on speed, coefficients for speed-dependent friction terms can be found.

krishauser reviewed (changes requested):
Do you have documentation on how to gather / generate these text files? Or scripts for gathering them? If not this is not useful for others.

krishauser reviewed (changes requested):
It is not enough to ask people in a comment to change their code to use this script. For this to count as a contribution you must provide code that works without manual editing, and can be triggered from a launch file

Andy-Chen-design commented:
Add acceleration calibration components and configuration

This commit introduces several updates and new files to enhance the acceleration calibration capabilities of our GEMstack:

1. **New Component**: `acc_calibration_driver.py` is introduced in `GEMstack/onboard/planning/`. This component facilitates the collection of vehicle dynamics data (measured acceleration, output acceleration, and speed) by running the vehicle at a constant acceleration specified via the driver.

2. **Utilities Update**: Modifications in `GEMstack/utils/acceleration_calibration.py` include new functionality to read data from a text file and perform curve fitting. This helps determine the friction coefficient for speed-dependent friction terms.

3. **Configuration File**: Added `launch/acceleration_calibration.yaml` which configures the vehicle to run at specified accelerations, thereby automating data collection for analysis.

4. **YAML Updates**: Updated `GEMstack/knowledge/defaults/computation_graph.yaml` to add a new component "driving_for calibration".

krishauser commented:
Rejected, low quality
